### PR TITLE
web-angular: reintroduce page object class based on pageloader 3

### LINKED ISSUE #517: web-angular: reintroduce basic component tests based on pageloader 3
The component tests were removed via:

- Angular: remove the dev-override on pageloader #487
- web-angular tests: drop use of page object class and pageloader mock #493

Related: https://github.com/dart-lang/site-webdev/issues/1351

### PR DESCRIPTION
**Pull request by chalin**
Fixes #517 

~@kevmoo - with this change, someone who creates a web-angular project in an IDE will need to run `webdev build` at least once, otherwise the analyzer will complain that `part 'app_po.g.dart'` doesn't exist.~

cc @kwalrath 

---

The updated tests are passing:

```nocode
00:00 +0: compiling test/app_test.dart
00:00 +0: test/app_test.dart: heading
00:00 +1: All tests passed!
```

### CODE REVIEW

**Comment by kevmoo** [comment]
@chalin check travis?

**Comment by chalin** [comment]
Ah, right, that issue of `app_po.g.dart` not being generated. Since that might change, I've simply committed the `app_po.g.dart` file for now.

**Comment by chalin** [comment]
Travis is happy now.

> simply committed the `app_po.g.dart` file for now.

Is that ok with you?

File: templates/web-angular/test/app_po.g.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
+
+part of 'app_po.dart';
+
+// **************************************************************************
+// Generator: PageObjectGenerator
+// **************************************************************************
+
+// ignore_for_file: private_collision_in_mixin_application
```
Comment (kevmoo) [inline templates/web-angular/test/app_po.g.dart]: Is this a bug somewhere? Or expected?

Comment (chalin) [reply templates/web-angular/test/app_po.g.dart]: Good question. @mk13?

File: templates/web-angular/test/app_po.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import 'package:pageloader/pageloader.dart';
+
+part 'app_po.g.dart';
+
+@PageObject()
+abstract class AppPO {
```
[the remark this answers is not in the document]
Comment (chalin) [reply templates/web-angular/test/app_po.dart]: See #519

**Pull request merged**